src/main/java/com/github/discvrseq/walkers/VcfToLuceneIndexer.java

```diff
@@ -0,0 +1,196 @@
+        // This will eventually be used if we need to implement custom InfoFieldAnnotation classes
+        return Collections.singletonList(new DiscvrVariantAnnotator.DiscvrAnnotationPluginDescriptor());
+    }
+
+    @Override
+    public boolean useVariantAnnotations() {
+        return true;
+    }
+
+    private final List<InfoFieldAnnotation> annotationsToUse = new ArrayList<>();
+
+    @Override
+    public void onTraversalStart() {
+        StandardAnalyzer analyzer = new StandardAnalyzer();
+
+        Directory index;
+        try {
+            index = FSDirectory.open(outDir.toPath());
+        } catch(IOException e) {
+            analyzer.close();
+            throw new GATKException(e.getMessage(), e);
+        }
+
+        IndexWriterConfig config = new IndexWriterConfig(analyzer);
+
+        try {
+            writer = new IndexWriter(index, config);
+        } catch (IOException e) {
+            throw new GATKException(e.getMessage(), e);
+        }
+
+        // This is where you'd sanity check the set of annotations is valid, etc.
+        // Note: we can implement custom InfoFieldAnnotation classes for operations like indexing all
+
+        // NOTE: as far as listing variable samples, see the GATK SampleList annotation
+        DiscvrVariantAnnotator.DiscvrAnnotationPluginDescriptor plugin = getCommandLineParser().getPluginDescriptor(DiscvrVariantAnnotator.DiscvrAnnotationPluginDescriptor.class);
+        for (String className : annotationClassNames) {
+            Class<?> clazz = plugin.getClassForPluginHelp(className);
+
+            try {
+                Annotation a = new DiscvrVariantAnnotator.DiscvrAnnotationPluginDescriptor().createInstanceForPlugin(clazz);
+                if (!(a instanceof InfoFieldAnnotation)) {
+                    throw new GATKException("All custom annotations must be InfoFieldAnnotation: " + className);
+                }
+                annotationsToUse.add((InfoFieldAnnotation) a);
+            }
+            catch (InstantiationException | IllegalAccessException e) {
+                throw new GATKException("Unable to create annotation: " + className);
+            }
+        }
+
+    }
+
+    private long sites = 0;
+
+    @Override
+    public void apply(VariantContext variant, ReadsContext readsContext, ReferenceContext referenceContext, FeatureContext featureContext) {
+        Map<String, Object> toIndex = new HashMap<>();
+
+        // Standard fields also included:
+        toIndex.put("contig", variant.getContig());
+        toIndex.put("start", variant.getStart());
+        toIndex.put("end", variant.getEnd());
+        // TODO: maybe others?
+
+
+        for (String infoField : infoFieldsToIndex) {
+            if (variant.hasAttribute(infoField) && variant.getAttribute(infoField) != null) {
+                toIndex.put(infoField, variant.getAttribute(infoField));
+            } else {
+                throw new GATKException("Non-existent INFO field key: " + infoField);
```

Comment:
I dont think this behavior is right. Having a null value shouldnt throw an exception. Annotation per-site can be sparse. We should instead do a check in onTraversalStart using the VcfHeader and INFO field data dictionary.